plugin/trino-memsql/src/main/java/io/trino/plugin/memsql/MemSqlTimeColumnMapping.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.memsql;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.trino.plugin.jdbc.ColumnMapping;
+import io.trino.plugin.jdbc.LongReadFunction;
+import io.trino.plugin.jdbc.LongWriteFunction;
+import io.trino.spi.type.TimeType;
+
+import java.time.LocalTime;
+import java.time.format.DateTimeParseException;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Verify.verify;
+import static io.trino.plugin.jdbc.StandardColumnMappings.fromTrinoTime;
+import static io.trino.spi.type.Timestamps.NANOSECONDS_PER_DAY;
+import static io.trino.spi.type.Timestamps.PICOSECONDS_PER_DAY;
+import static io.trino.spi.type.Timestamps.PICOSECONDS_PER_NANOSECOND;
+import static io.trino.spi.type.Timestamps.round;
+import static java.time.format.DateTimeFormatter.ISO_LOCAL_TIME;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * {@link MemSqlTimeColumnMapping} is a utility class for parsing times returned by {@code SingleStore} driver.
+ *
+ * {@see com.singlestore.jdbc.codec.list.LocalTimeCodec}
+ */
+class MemSqlTimeColumnMapping
```

Comment:
Inline all methods into MemSqlClient. All type handling should live there since the `toColumnMapping` and `toWriteMapping` are defined there (and these methods aren't expected to be re-used or tested directly).

Reply:
👍 this felt wrong, thanks for suggesting it.
Re: `these methods aren't expected to be re-used or tested directly` - we should revisit this - unit tests of these methods take < 1 second to run, don't startup Docker, etc, and they *should* be tested directly (IMO)

Reply:
Even if something works in a unit test it's possible it isn't plugged in correctly. After-all you will be providing hardcoded inputs to these functions which may or may not match what database returns. See also https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#avoid-mocks (while this isn't exactly a mock it does behave like one if you provide input values by hand)

Reply:
Right - mocking is garbage, but writing code than can be unit-tested w/out integration tests is #1 priority for me, and should be for all developers.
We couldn't survive without integration tests, but making things unit-testable is the foundation of everything else.

Reply:
File removed